src/services/DhService.ts

```diff
@@ -150,19 +167,16 @@ export abstract class DhService<TDH, TClient>
       this.outputChannel.appendLine('Creating session...');
       this.cachedCreateSession = this.createSession(this.dh, this.client);
 
-      const { cn = null, session = null } =
-        (await this.cachedCreateSession) ?? {};
+      try {
+        const { cn, session } = await this.cachedCreateSession;
 
-      // TODO: Use constant event name
-      if (cn != null) {
+        // TODO: Use constant 'disconnect' event name
```

Comment:
Should be able to get this constant from the API instance

Reply:
I'll need to think about how to refactor this one. As-is, DhService has `this.dh` as a generic type and is not aware of its members. This will probably become clearer as I start to add DHE which will require some refactoring anyway.